transform/transformers/ImageTransformer.py

```diff
@@ -1,43 +1,69 @@
-import os
-import glob
-import subprocess
-import zipfile
+#!/usr/bin/env python
+#   coding: UTF-8
+
+import argparse
 import datetime
 import dateutil.parser
-from transform import settings
-import shutil
+import glob
 from io import BytesIO
-from .PDFTransformer import PDFTransformer
-from transform.views.image_filters import get_env, format_date
+import itertools
+import json
+import logging
+import os.path
+import shutil
+import subprocess
+import sys
+import zipfile
+
 from requests.packages.urllib3.exceptions import MaxRetryError
+
+try:
+    from PDFTransformer import PDFTransformer
+except ImportError:
+    from .PDFTransformer import PDFTransformer
+
+from transform import settings
 from transform.settings import session
+from transform.views.image_filters import get_env, format_date
+
+__doc__ = """
+SDX Image Transformer.
+
+Example:
+
+python -m transform.transformers.ImageTransformer --survey transform/surveys/144.0001.json \\
+< tests/replies/ukis-01.json > output.zip
+
+"""
 
 
 class ImageTransformer(object):
-    def __init__(self, logger, survey, response_data, sequence_no=1000):
-        self.logger = logger
-        self.survey = survey
-        self.response = response_data
-        self.sequence_no = sequence_no
 
-    def create_pdf(self):
+    @staticmethod
+    def create_pdf(survey, data):
         '''
         Create a pdf which will be used as the basis for images
         '''
-        pdf_transformer = PDFTransformer(self.survey, self.response)
-
-        self.pdf_file = pdf_transformer.render_to_file()
-        self.path, self.base_name = os.path.split(self.pdf_file)
-        self.rootname, _ = os.path.splitext(self.base_name)
+        pdf_transformer = PDFTransformer(survey, data)
+        return pdf_transformer.render_to_file()
 
-    def extract_pdf_images(self):
+    @staticmethod
+    def extract_pdf_images(path, fileName):
         '''
         Extract all pdf pages as jpegs
         '''
-        subprocess.call(["pdftoppm", "-jpeg", self.base_name, self.rootname], cwd=self.path)
-        self.images = glob.glob("%s/%s-*.jpg" % (self.path, self.rootname))
+        rootName, _ = os.path.splitext(fileName)
+        subprocess.call(
+            ["pdftoppm", "-jpeg", fileName, rootName],
+            cwd=path
+        )
+        return glob.glob("%s/%s-*.jpg" % (path, rootName))
```

Comment:
Glob is returning files in arbitrary order (which may not be correct) - we've probably got away with this up until now as evorails may always order the files correctly (and we usually have a small number) but it seems ngd is not.

Probably need to explicitly sort the files before assigning them sequence numbers.